src/components/TimePicker/index.js

```diff
@@ -1,185 +1,149 @@
-import React, { Component } from 'react';
+/* eslint-disable react/no-unused-prop-types */
+import React, { useEffect, useRef, useState, useImperativeHandle } from 'react';
 import PropTypes from 'prop-types';
 import ClockIcon from './icons/clock';
 import Input from '../Input/pickerInput';
 import TimeSelect from './timeSelect';
 import get12HourTime from './helpers/get12HourTime';
 import getInputValue from './helpers/getInputValue';
-import withReduxForm from '../../libs/hocs/withReduxForm';
 import { ENTER_KEY, SPACE_KEY } from '../../libs/constants';
 import StyledContainer from './styled/container';
 import StyledModal from './styled/modal';
+import { useReduxForm } from '../../libs/hooks';
 
 /**
  * A TimePicker is used to input a time by displaying an interface the user can interact with.
  * @category Form
  */
-class TimePicker extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isOpen: false,
-            value: props.hour24 ? props.value : get12HourTime(props.value),
-        };
-        this.inputRef = React.createRef();
-        this.timeSelectRef = React.createRef();
-        this.handleKeyDown = this.handleKeyDown.bind(this);
-        this.handleClick = this.handleClick.bind(this);
-        this.closeModal = this.closeModal.bind(this);
-        this.setFocusToHourInput = this.setFocusToHourInput.bind(this);
-        this.handleBlur = this.handleBlur.bind(this);
-        this.handleFocus = this.handleFocus.bind(this);
-    }
-
-    componentDidUpdate({ value: prevValue }) {
-        const { value } = this.props;
-        if (prevValue !== value) {
-            this.updateValue();
-        }
-    }
-
-    getTriggerInputValue() {
-        const { value } = this.state;
-        const { placeholder, hour24 } = this.props;
+const TimePicker = React.forwardRef((props, ref) => {
+    const {
+        placeholder,
+        label,
+        required,
+        style,
+        className,
+        hideLabel,
+        name,
+        bottomHelpText,
+        isCentered,
+        error,
+        readOnly,
+        disabled,
+        tabIndex,
+        id,
+        cancelLabel,
+        okLabel,
+        onChange,
+        hour24,
+        onClick,
+        onBlur,
+        onFocus,
+    } = useReduxForm(props);
+    const [isOpen, setIsOpen] = useState(false);
+    const [value, setValue] = useState(hour24 ? props.value : get12HourTime(props.value));
```

Comment:
in case we use redux form we need to take the value prop from useReduxForm like all other props to get the right value